libcloud/common/gig_g8.py

```diff
@@ -26,6 +35,74 @@ def add_default_headers(self, headers):
         """
         Add headers that are necessary for every request
         """
+        self.driver.key = maybe_update_jwt(self.driver.key)
         headers['Authorization'] = "bearer {}".format(self.driver.key)
         headers['Content-Type'] = 'application/json'
         return headers
+
+
+def base64url_decode(input):
+    """
+    Helper method to base64url_decode a string.
+
+    :param input: Input to decode
+    :type  input: str
+
+    :rtype: str
+    """
+    rem = len(input) % 4
+
+    if rem > 0:
+        input += b'=' * (4 - rem)
+    return base64.urlsafe_b64decode(input)
+
+
+def is_jwt_expired(jwt):
```

Comment:
Can you please add some unit tests for this function (covering both expired and non expired scenario, etc.).

Reply:
I added a test for expire and none expired scenarios, not sure what to do about 'forms of input string' the method would pretty much crash with split or encoding errors if you pass invalid formats

Reply:
Thanks.

This should be fine for now.

With different forms of input strings, I meant thing such as JWT string payload missing some fields, being corrupted, etc. and making sure we handle that correctly.